### PR TITLE
Improved section's update CRUD and fixed issue that caused html pages to crash

### PR DESCRIPTION
**Pull request by Brodski**
I made all .html files (hopefully) unable to crash AND I prevented that issue where the console throws that error. The question mark you see does something like "If this is null, then just print null" Also, notice in the controler for professor, semester, and student. I changed the "Get" operation. I dont understand why that fixed it, but it did.

I made quite a few html pages to display which year & season a section belonged too.

I improved the update operation for section's room, to correctly prevent scheduling errors.

I overloaded the equals() method for section.java. I understand it looks a bit sketchy and it's possibly vulnerable to bugs, but I'm hoping we could improve it later. B/c for some reason I could not for my life of me remove a section from a list for some reason.

The Update on Section might be a bit buggy b/c the update dropbox doesn't exist yet :( but i put a place holder in the controller

I added a cute; 
boolean loadData = true;
    if (loadData) { // Load data" ..........
In that class that loads all the data

--->It shouldn't be crashing at all anymore! Yay!

### CODE REVIEW

File: src/main/java/unbanner/Section.java [hunk 1]
```diff
@@ -246,26 +246,72 @@ public void setId(ObjectId id) {
...
+      return true;
+    }
+    //if this begins before another section and ends after that section
+    if (this.time.getSecond().isBefore(sec.getTime().getFirst()) && this.time.getSecond().isAfter(sec.time.getSecond())) {
+      return true;
+    }
+    return false;
+  }
+
+  public boolean doesTimeConflictsRoom(){  // some ".compare" and ".equals" methods would have be ideal
```
Comment (cblanto7) [inline src/main/java/unbanner/Section.java]: Would you mind explaining the difference between this method and the already existing conflicts method?

File: src/main/java/unbanner/SectionController.java [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 import java.util.List;
 
+import groovy.transform.EqualsAndHashCode;
```
Comment (cblanto7) [inline src/main/java/unbanner/SectionController.java]: I believe that this was meant to be a lombok import

File: src/main/java/unbanner/StudentController.java [hunk 3]
```diff
@@ -88,8 +88,12 @@ public String newStudent(@ModelAttribute("student") Student student) {
   */
   @RequestMapping("/student/{id}")
   public String student(@PathVariable String id, Model model) {
-    model.addAttribute("student", repository.findOne(id));
-    return "student";
+    Student student = repository.findOne(id);
```
Comment (cblanto7) [inline src/main/java/unbanner/StudentController.java]: good job figuring this out

File: src/main/resources/templates/course.html [hunk 4]
```diff
@@ -9,7 +9,7 @@
 </head>
 <body>
     <div layout:fragment="content">
-        <h1>Course</h1>
+        <h1>Course  <span th:text="${course.name}"></span> </h1>
```
Comment (cblanto7) [inline src/main/resources/templates/course.html]: consider this line instead:
`<h1 th:text="${course.name}">course_name</h1>`
thymeleaf will replace 'course_name' with the value of course.name

**Review by cblanto7 — COMMENTED** [review]
I think you need to close this PR and resubmit it to try and solve the travis-ci issues. 

however, I cloned this repo and ran the gradle check task and there is one failing test. So it wont pass even if travis-ci was working.

This PR overall looks good to me, figured out some interesting stuff. There are some items in here I have questions about and want to address before we merge this. Please fix the broken test and then close and resubmit this pull request (to try and restart travis-ci) then I will do a full review and we can try and merge it to the dev branch

File: src/main/resources/templates/course.html (hunk 4 above)
Comment (Brodski) [reply src/main/resources/templates/course.html]: my bad, yeah that would have been better

**Comment by Brodski** [comment]
The reason I commented out the lines at creat_professor:

1) Even if I did fix it we would have to make a number of fixes to that html and the Create-button operation. The problem is that it's showing every section ever, instead of sections from the semester. 

2) It's also not a required feature by the costumer, it would be more important to work on fixing updates and deletes, ect.

3) I wasnt able to fix it in a in a reasonable time frame.

4) We're going to present Casey's noTestData version.

**Pull request closed without merging**